lib/services/usagereporter.go

```diff
@@ -253,3 +294,133 @@ func ConvertUsageEvent(event *usageevents.UsageEventOneOf, identityUsername stri
 		return nil, trace.BadParameter("invalid usage event type %T", event.GetEvent())
 	}
 }
+
+type TeleportUsageReporter struct {
+	// usageReporter is an actual reporter that batches and sends events
+	usageReporter *usagereporter.UsageReporter[prehogapi.SubmitEventRequest]
+	// anonymizer is the anonymizer used for filtered audit events.
+	anonymizer utils.Anonymizer
+	// clusterName is the cluster's name, used for anonymization and as an event
+	// field.
+	clusterName types.ClusterName
+	clock       clockwork.Clock
+}
+
+func (tur *TeleportUsageReporter) SubmitAnonymizedUsageEvents(events ...UsageAnonymizable) error {
```

Comment:
Since the input events are not yet anonymized, I'd consider renaming this. Maybe `AnonymizeAndSubmit(events...)`

Reply:
Agree, this name was slightly confusing for me in the original implementation, but I didn't change it. I will do it.